### DISCUSSION TITLE
EZ OpenCV for Dummies

### DISCUSSION THREAD

**Discussion by baqwas** [FIRST-Tech-Challenge/FtcRobotController#784] [Q&A] [answered]
I've kicked the tires with OpenCV under Android using pre-built libraries:

```
import org.opencv.android.OpenCVLoader;
import org.opencv.android.BaseLoaderCallback;
import org.opencv.android.CameraBridgeViewBase.CvCameraViewFrame;
import org.opencv.android.LoaderCallbackInterface;
import org.opencv.android.OpenCVLoader;
import org.opencv.core.Core;
import org.opencv.core.CvType;
import org.opencv.core.Mat;
import org.opencv.core.MatOfPoint;
import org.opencv.core.Rect;
import org.opencv.core.Scalar;
import org.opencv.core.Size;
import org.opencv.android.CameraBridgeViewBase;
import org.opencv.android.CameraBridgeViewBase.CvCameraViewListener2;
import org.opencv.imgproc.Imgproc;

   private CameraBridgeViewBase mOpenCvCameraView;

```

Is there a starter Hello EZOpenCV OpMode to leverage some simple image processing exercise(s) with the current VisionPortal processing? (I've done some reading of older threads but my comprehension skills are limited). Thanks.

Regards.

**Comment by Windwoes** [comment, marked as answer]
The Learn Java for FTC book has some examples for it in the vision chapter.

**Reply by baqwas** [reply]
Thanks!

**Comment by WestsideRobotics** [comment]
Just want to second the comment by @Windwoes.

In [**that book**](https://github.com/alan412/LearnJavaForFTC), start with section 16.2, and work your way carefully through section 16.5.  

You will build your own **OpenCV vision processor** (not AprilTag or TFOD), under **FTC VisionPortal**.

You will end up having, and understanding, a solution that works very well for CENTERSTAGE.

**Reply by baqwas** [reply]
I want our team to build some knowledge/experience with OpenCV and the _randomization_ challenge would be good exercise for the two lead programmers. We'll revert to AprilTags for navigation thanks to the sample OpMode. TFOD is out because our team seems to have less patience with model building. Thanks again.

Regards.

**Comment by baqwas** [comment]
@WestsideRobotics & @Windwoes, ran the first (of many tests) without any issue except that I will have to tweak the exposure settings _again_ (since I didn't copy anything from the previous exercises).

I too have scars building OpenCV from source since 2.7 (not for Android, of course) so I'm grateful for this FTC version of the library.

Regards.
